logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/backup/BackupMetadata.kt

```diff
@@ -0,0 +1,28 @@
+package com.wire.kalium.logic.feature.backup
+
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+@Serializable
+data class BackupMetadata(
+    @SerialName("platform")
+    val platform: String,
+    @SerialName("version")
+    val version: String,
+    @SerialName("user_id")
+    val userId: String,
+    @SerialName("creation_time")
+    val creationTime: String,
+    @SerialName("client_id")
+    val clientId: String
+) {
+    fun toMap(): Map<String, String> {
+        return mapOf(
+            "platform" to platform,
+            "version" to version,
+            "user_id" to userId,
+            "creation_time" to creationTime,
+            "client_id" to clientId
+        )
+    }
```

Comment:
not really clear what this should do, I see that u are using it to do this : 
.toMap().toJsonObject().toString()

would it make sense to declare a function toJsonObject here ? or something else that would do what you want ? also isn't toString() on data class doing what you want ?

Reply:
I added a `toMap()` function to be a bit coherent with what we are doing with other data class objects that are just a collection of key-value tuples. My main goal here was to be compliant with the requirements for the previous [exporting history pages](https://github.com/wearezeta/documentation/blob/0180afa7073909aef1dd199cd48fcddf4d915038/topics/backup/README.md#metadata-file). 
<img width="388" alt="Captura de Pantalla 2022-11-28 a las 15 38 01" src="https://user-images.githubusercontent.com/2468164/204304764-2e25ea2f-ba01-43b3-96ff-7b4b37f9a601.png">

But maybe I'm overthinking it and it could be done in a more simpler way. Any suggestions?

Reply:
makes sense I am not sure at the moment I was more relating to a fact that later u are doing this : 

val metadataJson = BackupMetadata(CLIENT_PLATFORM, BackupCoder.version, userId.toString(), creationTime, clientId.toString())
            .toMap().toJsonObject().toString()
            
`  .toMap().toJsonObject().toString()`
           could be move here instead of calling a toMap().toJsonObject().toString()